src/app/modules/glossary/glossary.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, OnInit, HostListener, ViewChild, Output, EventEmitter } from '@angular/core';
+import { Apollo, QueryRef } from 'apollo-angular';
+import { filter, map } from 'rxjs/operators';
+import { GlossaryWord } from './glossary.graphql';
+import { LinearLoaderService } from '../../shared/components/linear-loader/linear-loader.service';
+import { ErrorService } from 'src/app/shared/components/error-modal/error.service';
+import { UserService } from 'src/app/shared/services/user.service';
+import { ConsultationsService } from 'src/app/shared/services/consultations.service';
+import { ModalDirective } from 'ngx-bootstrap';
+
+@Component({
+  selector: 'app-glossary',
+  templateUrl: './glossary.component.html',
+  styleUrls: ['./glossary.component.scss']
+})
+
+export class GlossaryComponent implements OnInit {
+  @ViewChild('glossaryModal', { static: false }) glossaryModal: ModalDirective;
+  @Output() close: EventEmitter<any> = new EventEmitter();
+
+  glossaryWordArray: Array<any>;
+  glossaryWordQuery: QueryRef<any>;
+  currentUser: any;
+  consultationId: number;
+  showModal = true;
+  
+  constructor(
+    private apollo: Apollo,
+    private loader: LinearLoaderService,
+    private errorService: ErrorService,
+    private userService: UserService,
+    private consultationService: ConsultationsService,
+  ) {
+    this.consultationService.consultationId$
+      .pipe(
+        filter(i => i !== null)
```

Comment:
@gpalsingh - I guess a filter on the BE for consultation ID would avoid this?

Reply:
@mikevic This is to check for the [initial state](https://github.com/civis-vote/civis-frontend/blob/master/src/app/shared/services/consultations.service.ts#L11) i.e. Before the frontend finishes fetching data.

Reply:
The value itself seems to be coming from the [query parameters](https://github.com/civis-vote/civis-frontend/blob/master/src/app/modules/consultations/consultation-profile/consultation-profile.component.ts#L38). Perhaps the check can be moved to this piece of code instead